### ISSUE TITLE
Paket is very slow with ASP.NET Core 2

### ISSUE THREAD

**Issue by skalinets** [fsprojects/Paket#3126]
### Description

Performance is suboptimal when paket is used with ASP.NET Core 2. Just wondering if that is because of microsoft.aspnetcore.all metapackage or issue is the same as in #2827?

### Repro steps

1. `dotnet new webapi`

2. download paket.exe (ver  5.153.0), save it to `./paket` directory

3. `.\.paket\paket.exe convert-from-nuget`  takes ~ 5 minutes

4. `.\.paket\paket.exe update`  takes  minutes

5. `.\.paket\paket.exe add xunit`  takes minutes

Repro is available [here](https://github.com/skalinets/paket-issue)

### Expected behavior

It is expected to be much faster.

### Actual behavior

Steps 3..5 should not take minutes, but rather seconds. Console output can be observed in [my twitter](https://twitter.com/skalinets/status/957300242734514177). 

### Known workarounds

No any.

**Comment by skalinets** [comment]
@forki are there any workarounds or optimizations for paket? Currently it is almost unusable with ASP.NET Core projects. Maybe result of `convert-from-nuget` should be tweaked somehow?

**Comment by forki** [comment]
unsuable? I use it daily. what's the concrete issue?

**Comment by forki** [comment]
did you try to run "paket simplify"?

**Comment by forki** [comment]
did you remove version pins from the deps file?

**Comment by skalinets** [comment]
@forki the issue is that any 'paket restore' or 'paket add' take minutes, writting lots of nuget packages to the console ([here](https://twitter.com/skalinets/status/957300242734514177) is how it looks). Issue description contains link to the repo where it reproduces. [deps file](https://github.com/skalinets/paket-issue/blob/master/paket.dependencies) has just 2 dependencies. 

I tried to remove version pins as well as `paket simplify` -- it does not help. 

Probably I am doing something wrong, could you point me to so some repo with ASP.NET Core 2 where paket works as it is supposed to?

**Comment by forki** [comment]
You can look at https://github.com/SAFE-Stack/SAFE-BookStore

Let's talk about restore - what do you think it is doing wrong?

Serhiy Kalinets <notifications@github.com> schrieb am Do., 3. Mai 2018,
17:00:

**Comment by skalinets** [comment]
When I try to do `paket.exe add xunit` (just example, any other nuget package can be used here), it takes more than one minute. Even at SAFE-BookStore. Is it expected?

**Comment by forki** [comment]
Yes because that's an update process. It needs to resolve everything. And
asp.net ships A LOT of stuff.

Serhiy Kalinets <notifications@github.com> schrieb am Do., 3. Mai 2018,
19:07:

**Comment by forki** [comment]
Tbf there are a couple of things that we could do to speed up add command.
Anyway can you please post your deps file? Maybe I can spot something specific in your case

**Comment by skalinets** [comment]
it's just default file :)

```
source https://www.nuget.org/api/v2
nuget Microsoft.AspNetCore.All
clitool Microsoft.VisualStudio.Web.CodeGeneration.Tools 
```

**Comment by forki** [comment]
`Microsoft.AspNetCore.All` - that's a lot of stuff.

**Comment by skalinets** [comment]
I understand. But if I even change it to Microsoft.AspNetCore, it does not help too much. Now my deps file is 

```
source https://www.nuget.org/api/v2
nuget Microsoft.AspNetCore
nuget Microsoft.AspNetCore.Mvc
clitool Microsoft.VisualStudio.Web.CodeGeneration.Tools 
```

And `paket.exe add xunit` still takes 45 seconds. 

So if I want to add a couple of dependencies and don't want to wait a minute for each, the only option I see is doing it like

```
paket.exe add dep1 --no-resolve
paket.exe add dep2 --no-resolve
...
paket.exe install
```

**Comment by forki** [comment]
or you go to deps file. add it and run install

**Comment by forki** [comment]
I think I now understand your issue. You are not complaining about the resolution time, but you want to add couple of packages and resolve only once. is that correct?
